webrender/src/util.rs

```diff
@@ -92,6 +103,40 @@ impl<Src, Dst> MatrixHelpers<Src, Dst> for TypedTransform3D<f32, Src, Dst> {
             TransformedRectKind::Complex
         }
     }
+
+    // A port of the computeTypeMask() function from Skia.
+    fn compute_flags(&self) -> MatrixFlags {
+        let mut mask = MatrixFlags::empty();
+
+        if self.m14 != 0.0 ||
+           self.m24 != 0.0 ||
+           self.m34 != 0.0 ||
+           self.m44 != 1.0 {
+            return MatrixFlags::TRANSLATION |
+                   MatrixFlags::SCALE |
+                   MatrixFlags::AFFINE |
+                   MatrixFlags::PERSPECTIVE;
```

Comment:
can we really say the matrix has translation/scale if we only check for the perspective components here?